bin/parallel.py

```diff
@@ -31,15 +33,27 @@ def __init__(self, f, num_threads=True):
         self.finish = False
 
         if self.num_threads:
+            if pin:
+                # only use available cores
+                cores = list(os.sched_getaffinity(0))
+                if self.num_threads > len(cores):
+                    self.num_threads = len(cores)
```

Comment:
do 1 less to give some space to the OS. could even pin bapctools itself to that thread, but maybe not needed.